keyring/file.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package keyring
+
+import (
+	"crypto/sha1"
+	"os"
+	"path/filepath"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwe"
+	"github.com/zalando/go-keyring"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+var _ Keyring = (*fileKeyring)(nil)
+
+// fileKeyring is a keyring that stores keys in encrypted files.
+type fileKeyring struct {
+	// dir is the keystore root directory
+	dir string
+	// key is the keystore encryption key
+	key []byte
+}
+
+func openFileKeyring(dir string, password []byte) (*fileKeyring, error) {
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return nil, err
+	}
+	key := pbkdf2.Key(password, []byte("defradb"), 4096, 32, sha1.New)
```

Comment:
todo: Shouldn't be using a static salt here. I believe if you use the `jwa.PBES2_HS512_A256KW` instead of `jwa.A256KW` here then the `Encrypt` and `Decrypt` calls will handle the KDF and salt for you, and the salt then gets included in the JWE headers.

Reply:
then the `key` is just the `password` defined by the user, and you don't have to do the KDF manually.

Reply:
and the added benefit that it will use `sha256` instead of `sha1`

Reply:
I was hoping there was something like this in the library. I've replaced it as suggested.